pkg/api/requestcontext.go

```diff
@@ -89,21 +91,21 @@ func (c *requestContextMap) remove(req *http.Request) {
 	delete(c.contexts, req)
 }
 
-// NewRequestContextFilter ensures there is a Context object associated with the request before calling the passed handler.
+// WithRequestContext ensures there is a Context object associated with the request before calling the passed handler.
 // After the passed handler runs, the context is cleaned up.
-func NewRequestContextFilter(mapper RequestContextMapper, handler http.Handler) (http.Handler, error) {
-	if mapper, ok := mapper.(*requestContextMap); ok {
-		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			if mapper.init(req, NewContext()) {
-				// If we were the ones to successfully initialize, pair with a remove
-				defer mapper.remove(req)
-			}
-			handler.ServeHTTP(w, req)
-		}), nil
-	} else {
-		return handler, errors.New("Unknown RequestContextMapper implementation.")
+func WithRequestContext(handler http.Handler, mapper RequestContextMapper) http.Handler {
```

Comment:
Any idea why it's in this package?  Doesn't have to be this pull, but unless there's a strong reason, I'd like it moved eventually.